parser.py

```diff
@@ -0,0 +1,118 @@
+class StrTableParser(object):
+    """
+    Parser of string representation tables
+
+    Example of basic usage:
+    >>> StrTableParser(table=f'| head |\\n| body_1 |\\n| body_2 |').container
+    {'head': ['body_1', 'body_2']}
+    """
+
+    def __init__(self, table=None):
+        self._container = {}
+        self._table = None
+        self._fields = None
+        self._body = None
+
+        if table:
+            self.table = table
```

Comment:
The reason this check has been done is because 'table' is optional in constructor, so usage can be both: `StrTableParser().table(str_table_representation)` as well as `StrTableParser(str_table_representation)`.